exchanges/orderbook/linked_list.go

```diff
@@ -340,6 +345,74 @@ func (ll *linkedList) insertUpdates(updts Items, stack *stack, comp comparison)
 	return nil
 }
 
+// getSlippageByVolume returns the slippage percentage by impact volume on
+// liquidity.
+func (ll *linkedList) getSlippageByVolume(volume float64) (float64, error) {
+	if volume <= 0 {
+		return 0, errAmountInvalid
+	}
+	var slippage, shiftedPrice float64
+	for tip := &ll.head; *tip != nil; tip = &(*tip).Next {
+		shiftedPrice = (*tip).Value.Price
+		if slippage == 0 {
+			if shiftedPrice == 0 {
+				return 0, fmt.Errorf("invalid orderbook tranche %w", errPriceNotSet)
+			}
+			slippage = shiftedPrice
+		}
+		volume -= (*tip).Value.Amount
+		if volume <= 0 {
+			break
+		}
+	}
+	if shiftedPrice == 0 {
+		return 0, errNoLiquidity
+	}
+	if volume > 0 {
+		// Full book wiped out, return 100 percent.
+		return 100, nil
+	}
+	percentageChange := math.CalculatePercentageGainOrLoss(shiftedPrice, slippage)
```

Comment:
You should instead be using the average price across different items as per:

```
func TestDepthGetSlippageByVolume2(t *testing.T) {
	t.Parallel()
	d := NewDepth(id)

	asks := Items{
		{
			Price:  10000,
			Amount: 2,
		}, {
			Price:  10100,
			Amount: 7,
		}, {
			Price:  10200,
			Amount: 3,
		},
	}

	d.LoadSnapshot(bid, asks, 0, time.Time{}, true)

	s, err := d.GetSlippageByVolume(10, false)
	if err != nil {
		t.Fatal(err)
	}
	t.Log(s)

	b := Base{
		Asks: asks,
	}
	r1, err := b.GetAveragePrice(true, 10)
	if err != nil {
		t.Fatal(err)
	}

	s2 := (r1 - asks[0].Price) / asks[0].Price
	t.Log(s2 * 100)
}
```

This gives a more accurate slippage value of 0.89% instead of 2%.

I'd also like to be able to toggle between quote and base slippage calculations instead of just base amounts